torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -546,10 +546,24 @@ def __init__(
     ):
         torch._C._log_api_usage_once("torch.distributed.fsdp")
         super().__init__()
-        # Save the ignored modules and their parameters, including the
-        # parameter names, which are needed to filter the model state dict
-        self._ignored_modules = self._get_ignored_modules(ignored_modules)
-        ignored_params = self._get_ignored_parameters()
+        # Validate the ignored modules and derive the ignored parameters and
+        # buffers
+        ignored_modules = self._get_ignored_modules(module, ignored_modules)
```

Comment:
Maybe add a comment / warning in docstring in addition to this just to make sure it is completely clear to user that things might break if the parameter name gets swapped out underneath?

Reply:
Added a warning in the docstring.